asset_dashboard/models.py

```diff
@@ -300,6 +370,76 @@ class LocalAsset(models.Model):
     asset_model = models.CharField(max_length=100)
     asset_name = models.CharField(max_length=600)
 
+    @classmethod
+    def get_distribution_by_zone(cls, phase_geoms: GEOSGeometry) -> dict:
+        distributions_by_zone = {}
+
+        for zone in Zone.objects.all():
+            if zone.boundary:
+                intersection = zone.boundary.intersection(phase_geoms)
+                distributions_by_zone[zone] = intersection.area
+
+        return distributions_by_zone
+
+    @classmethod
+    def get_aggregated_assets_by_phase(cls, phase: Phase) -> GeometryCollection:
+        phase_assets = LocalAsset.objects.filter(phase=phase)
+
+        phase_polygons = LocalAsset.aggregate_polygons(phase_assets)
+        phase_linestrings = LocalAsset.aggregate_linestrings(phase_assets)
+        phase_points = LocalAsset.aggregate_points(phase_assets)
+
+        geoms = (
+            phase_polygons,
+            phase_linestrings,
+            phase_points
+        )
+
+        # filter out None
+        filtered_geoms = tuple([geom for geom in filter(None, geoms)])
+
+        return GeometryCollection(filtered_geoms)
+
+    @classmethod
+    def aggregate_polygons(cls, qs: QuerySet) -> Union[GEOSGeometry, None]:
+        assets = qs.extra(where=["""
+                                    geometrytype(geom) LIKE 'POLYGON'
+                                        OR geometrytype(geom) LIKE 'MULTIPOLYGON'
+                                 """]).aggregate(models.Union('geom'))['geom__union']
+
+        if assets:
+            return assets
+
+        return None
+
+    @classmethod
+    def aggregate_linestrings(cls, qs: QuerySet) -> Union[GEOSGeometry, None]:
+        assets = qs.extra(
+            where=["""
+                    geometrytype(geom) LIKE 'LINESTRING'
+                        OR geometrytype(geom) LIKE 'MULTILINESTRING'
+                   """]).aggregate(models.Union('geom'))['geom__union']
+
+        if assets:
+            # Return a buffered geometry so we can use the area of the buffer.
+            # The Geometries are in degrees, so use the 6th decimal place for
+            # creating the buffer. See https://gis.stackexchange.com/a/8674
+            # and https://docs.djangoproject.com/en/3.1/ref/contrib/gis/geos/#django.contrib.gis.geos.GEOSGeometry.buffer
+            return assets.buffer(.000005)
```

Comment:
let's make these buffer amounts settings that can be adjusted in the settings.py file.